app/models/course.rb

```diff
@@ -96,6 +96,10 @@ def available_study_modes_from_options
     course_options.select(&:site_still_valid).pluck(:study_mode).uniq
   end
 
+  def available_study_modes_with_vacancies
+    course_options.available.selectable.pluck(:study_mode).uniq
```

Comment:
We can just do `course_options.available`, as the `available` scope already uses `selectable`.

Reply:
Thanks for pointing out. Will have a look in a bit.